flang/lib/Lower/OpenMP.cpp

```diff
@@ -306,6 +306,35 @@ genOMP(Fortran::lower::AbstractConverter &converter,
   }
 }
 
+static mlir::omp::ClauseScheduleModifier
+translateModifier(const Fortran::parser::OmpScheduleModifierType &m) {
+  switch (m.v) {
+  case Fortran::parser::OmpScheduleModifierType::ModType::Monotonic:
+    return mlir::omp::ClauseScheduleModifier::Monotonic;
+  case Fortran::parser::OmpScheduleModifierType::ModType::Nonmonotonic:
+    return mlir::omp::ClauseScheduleModifier::Nonmonotonic;
+  default:
+    llvm_unreachable("Unknown case");
+  }
+  return mlir::omp::ClauseScheduleModifier::None;
+}
+
+static mlir::omp::ClauseScheduleModifier
+getScheduleModifiers(const Fortran::parser::OmpScheduleClause &x) {
+  const auto &modifier =
+      std::get<std::optional<Fortran::parser::OmpScheduleModifier>>(x.t);
+  if (modifier) {
+    const auto &modType1 =
+        std::get<Fortran::parser::OmpScheduleModifier::Modifier1>(modifier->t);
+    // TODO: Add support for SIMD, which means modType2 gets used.
```

Comment:
Should this be a hard `TODO()` here? (See flang/lower/Todo.h..) In general, we prefer that flang exits rather than emits incorrect code silently.

Reply:
As far as I understand it,it won't generate incorrect code [functionality-wise] by skipping the SIMD modifier. It may be less efficient than the intent from the programmer... :)